core/ledger/kvledger/txmgmt/privacyenabledstate/db_test.go

```diff
@@ -588,6 +588,56 @@ func generateLedgerID(t *testing.T) string {
 	return fmt.Sprintf("x%s", hex.EncodeToString(bytes))
 }
 
+func TestDrop(t *testing.T) {
+	for _, env := range testEnvs {
+		t.Run(env.GetName(), func(t *testing.T) {
+			testDrop(t, env)
+		})
+
+		t.Run("test-drop-error-propagation", func(t *testing.T) {
+			env.Init(t)
+			defer env.Cleanup()
+			ledgerid := generateLedgerID(t)
+			env.GetDBHandle(ledgerid)
+			env.GetProvider().Close()
+			require.EqualError(t, env.GetProvider().Drop(ledgerid), "internal leveldb error while obtaining db iterator: leveldb: closed")
+		})
+	}
+}
+
+func testDrop(t *testing.T, env TestEnv) {
+	env.Init(t)
+	defer env.Cleanup()
+	ledgerid := generateLedgerID(t)
+	db := env.GetDBHandle(ledgerid)
+
+	updates := NewUpdateBatch()
+	updates.PubUpdates.Put("ns1", "key1", []byte("value1"), version.NewHeight(1, 1))
+	putPvtUpdates(t, updates, "ns1", "coll1", "key1", []byte("pvt_value1"), version.NewHeight(1, 2))
+	db.ApplyPrivacyAwareUpdates(updates, version.NewHeight(2, 2))
+
+	vv, err := db.GetState("ns1", "key1")
+	require.NoError(t, err)
+	require.Equal(t, &statedb.VersionedValue{Value: []byte("value1"), Version: version.NewHeight(1, 1)}, vv)
+
+	vv, err = db.GetPrivateData("ns1", "coll1", "key1")
+	require.NoError(t, err)
+	require.Equal(t, &statedb.VersionedValue{Value: []byte("pvt_value1"), Version: version.NewHeight(1, 2)}, vv)
+
```

Comment:
can add an additional check here and below for `db.GetPrivateDataHash...` as well.